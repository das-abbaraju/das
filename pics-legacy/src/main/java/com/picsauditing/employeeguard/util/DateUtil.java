package com.picsauditing.employeeguard.util;

import com.picsauditing.employeeguard.entities.IntervalType;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

public class DateUtil {
	private static final Logger LOG = LoggerFactory.getLogger(DateUtil.class);

	public static final Date END_OF_TIME = new Date(64060617600000l);

	public static Date explodedToDate(final int year, final int month, final int day) {
		if (year > 0 && month > 0 && day > 0) {
			Calendar calendar = Calendar.getInstance();
			calendar.set(year, month - 1, day);
			return calendar.getTime();
		}

		return null;
	}

	public static int[] dateToExploded(final Date date) {
		if (date != null) {
			int[] exploded = new int[3];
			Calendar calendar = Calendar.getInstance();
			calendar.setTime(date);

			exploded[0] = calendar.get(Calendar.YEAR);
			exploded[1] = calendar.get(Calendar.MONTH) + 1;
			exploded[2] = calendar.get(Calendar.DAY_OF_MONTH);

			return exploded;
		}

		return new int[]{};
	}

	public static boolean doesNotExpire(final Date date) {
		if (date == null) {
			return false;
		}

		if (END_OF_TIME.equals(date)) {
			return true;
		}

		Calendar dayBeforeEndOfTime = Calendar.getInstance();
		dayBeforeEndOfTime.setTime(END_OF_TIME);
		dayBeforeEndOfTime.add(Calendar.DAY_OF_YEAR, -1);

		return date.after(dayBeforeEndOfTime.getTime());
	}

	public static boolean isAboutToExpire(final Date documentExpirationDate, final int intervalPeriod, final IntervalType intervalType) {
		Calendar documentExpiration = Calendar.getInstance();
		documentExpiration.setTime(documentExpirationDate);

		switch (intervalType) {
			case DAY:
				documentExpiration.add(Calendar.DAY_OF_YEAR, -intervalPeriod);
				break;
			case WEEK:
				documentExpiration.add(Calendar.WEEK_OF_YEAR, -intervalPeriod);
				break;
			case MONTH:
				documentExpiration.add(Calendar.MONTH, -intervalPeriod);
				break;
			case YEAR:
				documentExpiration.add(Calendar.YEAR, -intervalPeriod);
				break;
			case NO_EXPIRATION:
				return false;
			case NOT_APPLICABLE:
				return false;
			default:
				throw new IllegalArgumentException();
		}

		return new Date().after(documentExpiration.getTime());
	}

  public static String getSystemTime() {
    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
    Date systemTime = new Date();

    return dateFormat.format(systemTime);
  }
}
