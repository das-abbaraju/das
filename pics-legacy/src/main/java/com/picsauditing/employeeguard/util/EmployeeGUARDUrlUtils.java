package com.picsauditing.employeeguard.util;

public final class EmployeeGUARDUrlUtils {

	/* Corporate and Operator Links */
	public static final String OPERATOR_SUMMARY = "/employee-guard/operators/dashboard";
	public static final String OPERATOR_ASSIGNMENTS = "/employee-guard/operators/assignments";
	public static final String OPERATOR_PROJECTS = "/employee-guard/operators/projects";
	public static final String OPERATOR_JOB_ROLES = "/employee-guard/operators/role";
	public static final String OPERATOR_SKILLS = "/employee-guard/operators/skill";
	public static final String OPERATOR_PROJECT_CREATE = "/employee-guard/operators/project/create";

	/* Contractor Links */
	public static final String CONTRACTOR_SUMMARY = "/employee-guard/contractor/dashboard";
	public static final String CONTRACTOR_ASSIGNMENTS = "/employee-guard/contractor/project";
	public static final String CONTRACTOR_GROUPS = "/employee-guard/contractor/employee-group";
	public static final String CONTRACTOR_SKILLS = "/employee-guard/contractor/skill";
	public static final String CONTRACTOR_EMPLOYEES = "/employee-guard/contractor/employee";
	public static final String CONTRACTOR_EMPLOYEE_IMPORT = "/contractor/employee/import-export";

	/* Employee Links */
	public static final String IMAGE_LINK = "/employee-guard/employee/contractor/%d/employee-photo/%d";
	public static final String EMPLOYEE_PROFILE = "/employee-guard/employee/profile/%d";
	public static final String EMPLOYEE_SUMMARY = "/employee-guard/employee/dashboard";
	public static final String EMPLOYEE_MY_FILES = "/employee-guard/employee/file";
	public static final String EMPLOYEE_SKILLS = "/employee-guard/employee/skills";

	public static String buildUrl(final String url, final Object... params) {
		return String.format(url, params);
	}

}
