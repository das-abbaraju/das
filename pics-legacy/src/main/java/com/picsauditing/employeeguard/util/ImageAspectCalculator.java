package com.picsauditing.employeeguard.util;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ImageAspectCalculator {

	/**
	 *
	 * Given scale=300
	 *
	 * Example 1
	 * 3000 X 1000
	 * new Width=300, new Height= (300X1000)/3000
	 *
	 * Example 2
	 * 1000 X 1000
	 * new Height=300, new Width= (300X1000)/3000
	 *
	 * Example 3
	 * 100 X 50
	 * Same values returned since both are < scale value
	 *
	 * @param scale
	 * @return
	 */
	public AspectResult calculateThumnailSize(int width, int height, int scale){
		int newWidth=0;
		int newHeight=0;

		if(width < scale && height < scale){
			return new AspectResult(width, height);
		}

		if(width ==0  || height == 0){
			return new AspectResult(width, height);
		}

		if(width > height){
			newWidth = scale;
			newHeight = (scale*height)/width;
		}
		else{
			newHeight = scale;
			newWidth =  (scale*width)/height;

		}

		return new AspectResult(newWidth, newHeight);
	}

	private class AspectResult{
		private int width;
		private int height;

		private AspectResult(int width, int height) {
			this.width = width;
			this.height = height;
		}

		public int getWidth() {
			return width;
		}

		public int getHeight() {
			return height;
		}
	}
}
