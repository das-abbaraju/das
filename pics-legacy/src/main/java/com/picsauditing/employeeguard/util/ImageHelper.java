package com.picsauditing.employeeguard.util;

import org.imgscalr.Scalr;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.*;

public class ImageHelper {

	public static final int THUMBNAIL_DEFAULT_SCALE=300;
	public static final String THUMBNAIL_DEFAULT_FORMAT="jpg";

	/**
	 *
	 * Given scale=300
	 *
	 * Example 1
	 * 3000 X 1000
	 * new Width=300, new Height= (300X1000)/3000
	 *
	 * Example 2
	 * 1000 X 1000
	 * new Height=300, new Width= (300X1000)/3000
	 *
	 * Example 3
	 * 100 X 50
	 * Same values returned since both are < scale value
	 *
	 * @param scale
	 * @return
	 */
	public static AspectResult calculateThumnailSize(int width, int height, int scale){
		int newWidth=0;
		int newHeight=0;

		if(width < scale && height < scale){
			return new AspectResult(width, height);
		}

		if(width ==0  || height == 0){
			return new AspectResult(width, height);
		}

		if(width > height){
			newWidth = scale;
			newHeight = (scale*height)/width;
		}
		else{
			newHeight = scale;
			newWidth =  (scale*width)/height;

		}

		return new AspectResult(newWidth, newHeight);
	}

	public static InputStream resizeImage(int width, int height, BufferedImage originalImage, String formatName) throws IOException {

		BufferedImage resizedImage = Scalr.resize(originalImage, Scalr.Method.BALANCED, width, height);

		ByteArrayOutputStream baos = new ByteArrayOutputStream();
		ImageIO.write(resizedImage, formatName, baos);
		InputStream is = new ByteArrayInputStream(baos.toByteArray());

		return is;
	}

	public static class AspectResult{
		private int width;
		private int height;

		private AspectResult(int width, int height) {
			this.width = width;
			this.height = height;
		}

		public int getWidth() {
			return width;
		}

		public int getHeight() {
			return height;
		}
	}
}
