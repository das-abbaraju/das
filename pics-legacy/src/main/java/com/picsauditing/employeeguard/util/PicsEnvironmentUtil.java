/**
 *
 *
 * User: aphatarphekar
 * Date: 7/17/2014
 * Time: 2:56 PM
 *
 */
package com.picsauditing.employeeguard.util;

import com.picsauditing.dao.AppPropertyDAO;
import com.picsauditing.jpa.entities.AppProperty;

import com.picsauditing.service.AppPropertyService;
import com.picsauditing.util.system.PicsEnvironment;
import org.springframework.beans.factory.annotation.Autowired;

public class PicsEnvironmentUtil {

	@Autowired
	private AppPropertyService appPropertyService;

	private PicsEnvironment picsEnvironment;

	public PicsEnvironmentUtil() {
	}

	private PicsEnvironment getPicsEnvInstance(){
		if(picsEnvironment==null)
			picsEnvironment = new PicsEnvironment(appPropertyService.getPropertyString(AppProperty.VERSION_MAJOR),
						appPropertyService.getPropertyString(AppProperty.VERSION_MINOR));

		return picsEnvironment;
	}

	public String getPicsEnvironment() {

		return getPicsEnvInstance().getEnvironment();
	}

	public boolean isLocalhostEnvironment() {
		return getPicsEnvInstance().isLocalhost();
	}
}

 