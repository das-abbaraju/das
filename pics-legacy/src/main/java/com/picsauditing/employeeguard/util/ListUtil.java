package com.picsauditing.employeeguard.util;

import java.util.Comparator;
import java.util.List;
import java.util.Set;
import java.util.TreeSet;

public class ListUtil {
	public static <E extends Comparable<E>> List<E> removeDuplicatesAndSort(List<E> list) {
		Set<E> unique = new TreeSet<>(list);
		list.clear();
		list.addAll(unique);

		return list;
	}

	public static <E> List<E> removeDuplicatesAndSort(List<E> list, Comparator<E> comparator) {
		Set<E> unique = new TreeSet<>(comparator);
		unique.addAll(list);
		list.clear();
		list.addAll(unique);

		return list;
	}
}
