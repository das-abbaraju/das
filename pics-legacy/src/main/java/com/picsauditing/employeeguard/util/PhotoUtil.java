package com.picsauditing.employeeguard.util;

import com.picsauditing.PICS.PICSFileType;
import com.picsauditing.employeeguard.entities.DocumentType;
import com.picsauditing.employeeguard.entities.Employee;
import com.picsauditing.employeeguard.entities.Profile;
import com.picsauditing.employeeguard.entities.ProfileDocument;
import com.picsauditing.util.FileUtils;

import java.io.File;

public class PhotoUtil {
	public static final String[] VALID_PHOTO_EXTENSIONS = new String[]{"jpg", "png", "gif"};

	public static void sendPhotoToFilesDirectory(File photo, String directory, int id, String extension, String filename) throws Exception {
		FileUtils.moveFile(photo, directory, "files/" + FileUtils.thousandize(id), filename, extension, true);
	}

	public static File getPhotoForProfile(final Profile profile, final String directory) {
		ProfileDocument photoDocument = getPhotoDocumentFromProfile(profile);
		if (photoDocument != null) {
			File file = new File(directory + "/files/" + FileUtils.thousandize(profile.getId()) + photoDocument.getFileName());
			if (file.exists()) {
				return file;
			}
		}

		return null;
	}

	public static File getDefaultPhoto(final String directory) {
		File file = new File(directory + "/files/dummy.jpg");
		if (file.exists()) {
			return file;
		}

		return null;
	}

	public static ProfileDocument getPhotoDocumentFromProfile(final Profile profile) {
		for (ProfileDocument profileDocument : profile.getDocuments()) {
			if (profileDocument.getDocumentType() == DocumentType.Photo) {
				return profileDocument;
			}
		}

		return null;
	}

	public static void deleteExistingProfilePhoto(String directory, Profile profile) {
		File file = getPhotoForProfile(profile, directory);
		if (file != null && file.exists()) {
			FileUtils.deleteFile(file);
		}
	}

	public static File getPhotoForEmployee(Employee employee, int accountId, String directory) {
		int id = employee.getId();
		String filename = PICSFileType.employee_photo.filename(id) + "-" + accountId;

		File file;
		for (String extension : VALID_PHOTO_EXTENSIONS) {
			file = new File(directory + "/files/" + FileUtils.thousandize(id) + filename + "." + extension);

			if (file != null && file.exists()) {
				return file;
			}
		}

		return null;
	}

	public static boolean isValidExtension(String extension) {
		return FileUtils.checkFileExtension(extension, VALID_PHOTO_EXTENSIONS);
	}

    public static File getFile(String directory, ProfileDocument document) {
        if (document == null) {
            return null;
        }

        File file = new File(directory + "/files/" + FileUtils.thousandize(document.getProfile().getId())
                + document.getFileName());

        if (file.exists()) {
            return file;
        }

        return null;
    }
}
