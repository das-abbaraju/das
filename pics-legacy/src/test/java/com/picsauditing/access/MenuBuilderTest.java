package com.picsauditing.access;

import com.picsauditing.PicsTranslationTest;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.powermock.reflect.Whitebox;

import java.util.Locale;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import static org.mockito.Matchers.any;
import static org.mockito.Matchers.eq;
import static org.mockito.Mockito.when;

public class MenuBuilderTest extends PicsTranslationTest {
	public static final int USER_ID = 123;
	public static final int SWITCHED_TO_ID = 456;

	@Mock
	private Permissions permissions;

	@Before
	public void setUp() {
		MockitoAnnotations.initMocks(this);

		when(permissions.getUserId()).thenReturn(USER_ID);
	}

	@Test
	public void testSwitchedToAnotherUser_AdminIdNotSet() throws Exception {
		boolean switchedToAnotherUser = Whitebox.invokeMethod(MenuBuilder.class, "switchedToAnotherUser", permissions);
		assertFalse(switchedToAnotherUser);
	}

	@Test
	public void testSwitchedToAnotherUser_AdminIdDoesNotEqualUserId() throws Exception {
		when(permissions.getAdminID()).thenReturn(USER_ID);
		when(permissions.getUserId()).thenReturn(SWITCHED_TO_ID);

		boolean switchedToAnotherUser = Whitebox.invokeMethod(MenuBuilder.class, "switchedToAnotherUser", permissions);
		assertTrue(switchedToAnotherUser);
	}

	@Test
	public void testSwitchedToAnotherUser_AdminIdEqualsUserId() throws Exception {
		when(permissions.getAdminID()).thenReturn(USER_ID);
		when(permissions.getUserId()).thenReturn(USER_ID);

		boolean switchedToAnotherUser = Whitebox.invokeMethod(MenuBuilder.class, "switchedToAnotherUser", permissions);
		assertFalse(switchedToAnotherUser);
	}

    @Test
    public void testGetMibewURL_LoggedIn() throws Exception {
        String expectedURL = "https://chat.picsorganizer.com/client.php?locale=en&style=PICS&name=Name&accountName=AccountName&accountId=12345&userId=54321&email=me%40example.com";
        when(translationService.getText(eq("Mibew.LanguageCode"), any(Locale.class))).thenReturn("en");
        setupPermissionsForMibew();

        String mbewURL = MenuBuilder.getMibewURL(Locale.ENGLISH, permissions);

        assert(expectedURL.equals(mbewURL));
    }

    @Test
    public void testGetMibewURL_NotLoggedIn() throws Exception {
        String expectedURL = "https://chat.picsorganizer.com/client.php?locale=en&style=PICS";
        when(translationService.getText(eq("Mibew.LanguageCode"), any(Locale.class))).thenReturn("en");
        when(permissions.isLoggedIn()).thenReturn(false);

        String mbewURL = MenuBuilder.getMibewURL(Locale.ENGLISH, permissions);

        assert(expectedURL.equals(mbewURL));
    }

    private void setupPermissionsForMibew() {
        when(permissions.isLoggedIn()).thenReturn(true);
        when(permissions.getName()).thenReturn("Name");
        when(permissions.getAccountName()).thenReturn("AccountName");
        when(permissions.getAccountId()).thenReturn(12345);
        when(permissions.getUserId()).thenReturn(54321);
        when(permissions.getEmail()).thenReturn("me@example.com");
    }
}
