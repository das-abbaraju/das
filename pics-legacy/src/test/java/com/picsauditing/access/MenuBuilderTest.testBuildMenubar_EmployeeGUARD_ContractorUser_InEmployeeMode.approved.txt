[MenuComponent[level=1,name=Translate[global.Menu.EmployeeGUARD.Employee.Profile=>en],url=/employee-guard/employee/profile/1,htmlId=employee_profile,title=<null>,target=<null>,id=100,sortField=<null>,auditId=0,current=false,cssClass=,dataFields={},children=[]], MenuComponent[level=1,name=Translate[global.Menu.EmployeeGUARD.Employee.Skills=>en],url=/employee-guard/employee/skills,htmlId=employee_skills,title=<null>,target=<null>,id=101,sortField=<null>,auditId=0,current=false,cssClass=,dataFields={},children=[]], MenuComponent[level=1,name=Translate[global.Menu.EmployeeGUARD.Employee.My_Files=>en],url=/employee-guard/employee/file,htmlId=employee_my_files,title=<null>,target=<null>,id=102,sortField=<null>,auditId=0,current=false,cssClass=,dataFields={},children=[]], MenuComponent[level=1,name=John Doe,url=<null>,htmlId=user_menu,title=<null>,target=<null>,id=103,sortField=<null>,auditId=0,current=false,cssClass=,dataFields={},children=[MenuComponent[level=2,name=Translate[global.Menu.EmployeeGUARD.User.Switch_To_Admin_Mode=>en],url=/users/switch-mode/admin,htmlId=<null>,title=<null>,target=<null>,id=10300,sortField=<null>,auditId=0,current=false,cssClass=,dataFields={},children=[]], MenuComponent[level=2,name=Translate[global.Menu.EmployeeGUARD.User.Settings=>en],url=/employee-guard/profile/settings,htmlId=<null>,title=<null>,target=<null>,id=10301,sortField=<null>,auditId=0,current=false,cssClass=,dataFields={},children=[]], MenuComponent[level=2,name=Translate[Header.Logout=>en],url=Login.action?button=logout,htmlId=logout,title=<null>,target=<null>,id=10302,sortField=<null>,auditId=0,current=false,cssClass=,dataFields={},children=[]]]]]